api/src/main/java/org/ccci/idm/user/Dn.java

```diff
@@ -41,20 +41,20 @@ public final String getName() {
     }
 
     public final boolean isDescendantOf(@Nonnull final Dn ancestor) {
-        return ancestor.components.size() < components.size() && Objects.deepEquals(ancestor.components, components.subList(0, ancestor
-                .components.size()));
+        return ancestor.components.size() < components.size() && isDescendantOfOrEqualTo(ancestor);
     }
 
     public final boolean isDescendantOfOrEqualTo(@Nonnull final Dn ancestor) {
-        return equals(ancestor) || isDescendantOf(ancestor);
```

Comment:
see this comment: https://github.com/CruGlobal/idm-user-management/pull/85#issuecomment-216843245

`.equals` is used internally for object uniqueness checks within Collections and other internal objects. Because of that I would argue that a `Group` object and a `Dn` object should never be the same. This would break the `isDescendantOfOrEqualTo` if you were comparing a `Group` to a `Dn` that represented the same underlying DN